### ISSUE TITLE
Watch older episodes 

### ISSUE THREAD

**Issue by AlexKalopsia** [y20k/escapepod#42]
Hi, 

thanks for making Escapepod. I added a podcast that I was catching up on and while I did read I can only pull 5 episodes, I'm wondering if it could be possible to somehow give it a timeframe for those episodes.

**Comment by y20k** [comment]
Hi @AlexKalopsia,

I think ultimately Escapepod should be able to download and play ANY episode of a given podcast. I decided to limit the number of episodes to 5 five due to performance issues and unsolved UI issues.

Performance: The app current stores the metadata of all podcasts/episodes in a single JSON file, that it reads from and writes to device storage. This file gets too large if you have unlimited numbers of episodes. I would need start using a database to handle this. I actually I am doing that already on the testing branch. Put my progress is due to time constraints not very fast here.

UI: Once the performance issues are solved, I need to think about the UI. The only solution I can think of would be a new "All Episodes (of Podcast X)" screen. Plus: Indicators that there are episodes downloaded somewhere in the "All Episodes of Podcast X" list. There also maybe the need to add a list of "All Downloads". 

If I would do that, Escapepod would become more and more a full fledged podcast client, like AntennaPod or Pocket Casts. I really like an often/sometimes use those apps myself. For example when I am trying to catch up with a podcast. :smile: But I intentionally built Escapepod for myself, because the big podcast apps were to complicated for me personally to use on a day to day basis. 

In short: There is a chance that I will enable access to all available episodes of a podcast. But only If I find a way not to compromise on simplicity. I am taking this one step at a time. And development is slow here, since I am doing this in my free time. For now I would suggest to use another podcast app.

**Comment by y20k** [comment]
I forgot that there is a related feature request: #18

**Comment by y20k** [comment]
I will reopen or comment on ... this Issue or Issue 18 when I am actually going to work on the feature
